src/core/portal/portal.spec.ts

```diff
@@ -0,0 +1,183 @@
+
+      builder.createAsync(PortalTestApp).then(fixture => {
+        appFixture = fixture;
+      });
+
+      // Flush the async creation of the PortalTestApp.
+      flushMicrotasks();
+
+      let testAppComponent = appFixture.debugElement.componentInstance;
+
+      // Detect changes initially so that the component's ViewChildren are resolved.
+      appFixture.detectChanges();
+
+      // Set the selectedHost to be a ComponentPortal.
+      testAppComponent.selectedPortal = testAppComponent.cakePortal;
+      appFixture.detectChanges();
+
+      // Flush the attachment of the Portal.
+      flushMicrotasks();
+
+      // Expect that the content of the attached portal is present.
+      let hostContainer = appFixture.nativeElement.querySelector('.portal-container');
+      expect(hostContainer.textContent).toContain('Cake');
+    }));
+
+    it('should load a <template> portal with the `*` sugar', fakeAsyncTest(() => {
+      let appFixture: ComponentFixture;
+
+      builder.createAsync(PortalTestApp).then(fixture => {
+        appFixture = fixture;
+      });
+
+      // Flush the async creation of the PortalTestApp.
+      flushMicrotasks();
+
+      let testAppComponent = appFixture.debugElement.componentInstance;
+
+      // Detect changes initially so that the component's ViewChildren are resolved.
+      appFixture.detectChanges();
+
+      // Set the selectedHost to be a ComponentPortal.
+      testAppComponent.selectedPortal = testAppComponent.piePortal;
+      appFixture.detectChanges();
+
+      // Flush the attachment of the Portal.
+      flushMicrotasks();
+
+      // Expect that the content of the attached portal is present.
+      let hostContainer = appFixture.nativeElement.querySelector('.portal-container');
+      expect(hostContainer.textContent).toContain('Pie');
+    }));
+
+    it('should change the attached portal', fakeAsyncTest(() => {
+      let appFixture: ComponentFixture;
+
+      builder.createAsync(PortalTestApp).then(fixture => {
+        appFixture = fixture;
+      });
+
+      // Flush the async creation of the PortalTestApp.
+      flushMicrotasks();
+
+      let testAppComponent = appFixture.debugElement.componentInstance;
+
+      // Detect changes initially so that the component's ViewChildren are resolved.
+      appFixture.detectChanges();
+
+      // Set the selectedHost to be a ComponentPortal.
+      testAppComponent.selectedPortal = testAppComponent.piePortal;
+      appFixture.detectChanges();
+
+      // Flush the attachment of the Portal.
+      flushMicrotasks();
+
+      // Expect that the content of the attached portal is present.
+      let hostContainer = appFixture.nativeElement.querySelector('.portal-container');
+      expect(hostContainer.textContent).toContain('Pie');
+
+      testAppComponent.selectedPortal = new ComponentPortal(PizzaMsg);
+      appFixture.detectChanges();
+
+      flushMicrotasks();
+
+      expect(hostContainer.textContent).toContain('Pizza');
+    }));
+
+  });
```

Comment:
Well, this sent me down a long, long journey of trying to get the test working for this. There _was_ a bug, which was that I had `k` and `v` swapped in my `forEach`.

However, there is triumph, and there's now  a unit test for the `locals` behavior as well as a test for a `TemplatePortal` that contains a binding.